llvm/include/llvm/Frontend/HLSL/RootSignatureMetadata.h

```diff
@@ -26,6 +28,80 @@ class Metadata;
 namespace hlsl {
 namespace rootsig {
 
+template <typename T>
+class RootSignatureValidationError
+    : public ErrorInfo<RootSignatureValidationError<T>> {
+public:
+  static char ID;
+  std::string ParamName;
+  T Value;
+
+  RootSignatureValidationError(StringRef ParamName, T Value)
+      : ParamName(ParamName.str()), Value(Value) {}
```

Comment:
Don't need to make a string here.
```suggestion
      : ParamName(ParamName), Value(Value) {}
```